lib/repl.js

```diff
@@ -1141,6 +1141,165 @@ REPLServer.prototype.convertToContext = function(cmd) {
   return cmd;
 };
 
+REPLServer.prototype.setupHistory = function(historyPath, oldHistoryPath, cb) {
```

Comment:
That makes sense. My first jab at this was to keep the existing functionality as similar as possible, in order to get the discussion going without making assumptions or rocking the boat. FWIW, in this PR, `oldHistoryPath` is optional, so it wouldn't be required by the end user. But yes, it's cleaner to not expose this.

Currently, this is user controlled after the REPL has been created - similar to how it is now in `createInternalRepl`. I suppose, exposing it to the user as a post-creation action does introduce the race potential.